dev_tools/mypy.ini

```diff
@@ -0,0 +1,27 @@
+[mypy]
+show_error_codes = true
+
+[mypy-__main__]
+follow_imports = silent
+ignore_missing_imports = true
+
+# 3rd-party libs for which we don't have stubs
+[mypy-apiclient.*,freezegun.*,matplotlib.*,mpl_toolkits,multiprocessing.dummy,oauth2client.*,pandas.*,proto.*,pytest.*,scipy.*,sortedcontainers.*,setuptools.*,pylatex.*,networkx.*,qiskit.*,pypandoc.*,ply.*,_pytest.*,google.api.*,google.api_core.*,grpc.*,google.auth.*,google.oauth2.*,google.protobuf.text_format.*,quimb.*,pyquil.*,google.cloud.*,filelock.*,codeowners.*,tqdm.*,importlib_metadata.*,google.colab.*,IPython.*,astroid.*,pylint.*,cirq.*,flask.*,numpy.*,cirq_google.*]
```

Comment:
Wow, do you really have to maintain a list of transitive dependencies to use mypy? I think this will be a problem (if the check is enforced). The set of dependencies can change at any time since the package manager generally downloads the latest versions modulo constraints. So this will lead to CI errors on pull requests unrelated to the actual changes whenever a new transitive dependency appears.

Reply:
It's not transitive dependencies, it's only direct dependencies.  CI errors for this will only be if a new dependency appears that was introduced in the actual PR.

Hopefully I can figure out how to get stubs for some of the libraries as well.
Let me see if I can clean this up.

Reply:
What do you mean it's not transitive dependencies? For example `freezegun`, `protobuf`, or `flask` are not referenced by the Unitary codebase.

Reply:
Those were all from cirq, where I copied the continuous integration test from.  So, there were a lot of extra unneeded dependencies in there.  If you refresh to the latest commit, I have pruned it way down (there may still be a few extra unneeded ones, but I got rid of a bunch of them).